app/credentials/env_credentials.rb

```diff
@@ -0,0 +1,173 @@
+require "concerns/user_provided_services"
+
+class EnvCredentials
+  extend UserProvidedService
+
+  def self.new_relic_app_name
+    if use_env_var?
+      ENV["NEW_RELIC_APP_NAME"]
+    else
+      credentials("NEW_RELIC_APP_NAME")
```

Comment:
See comment above - this assumes that we've got all credentials in a single UPS.